File: src/main/java/land/leets/domain/interview/presentation/dto/req/InterviewAttendanceRequest.java
package land.leets.domain.interview.presentation.dto.req;

import jakarta.validation.constraints.NotNull;
import land.leets.domain.interview.domain.Interview;
import land.leets.domain.interview.type.HasInterview;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.UUID;

@Getter
@AllArgsConstructor
public class InterviewAttendanceRequest {
    @NotNull
    private UUID uid;

    @NotNull
    private HasInterview hasInterview;

    public void updateInterview(Interview interview) {
        interview.setHasInterview(this.hasInterview);
    }
}

File: src/main/kotlin/land/leets/domain/interview/presentation/dto/req/InterviewAttendanceRequest.kt
package land.leets.domain.interview.presentation.dto.req

import jakarta.validation.constraints.NotNull
import land.leets.domain.interview.domain.Interview
import land.leets.domain.interview.type.HasInterview
import java.util.UUID

data class InterviewAttendanceRequest(
    @NotNull
    val uid: UUID,

    @NotNull
    val hasInterview: HasInterview
) {
    fun updateInterview(interview: Interview) {
        interview.hasInterview = this.hasInterview
    }
}
